src/Elements/types.jl

```diff
@@ -0,0 +1,180 @@
+const HEADERS   = ["#", "##", "###", "####", "#####", "######"]
+const EMPHASIS  = ["*", "**"]
+
+const CATEGORIES = [:module,    :function, :method, :macro, :type,
```

Comment:
Expanding on that:

If we have documented function ``foo`` and methods ``foo(a::Int)``, ``foo{T <: Real}(a::Int, b::Array{T, 3})`` then we get output something like:

---

```
foo
```

Documentation for the ``Function`` object...

```
foo(a :: Int)
```

Documentation for the ``Method`` object...

```
foo{
    T <: Real
}(
    a :: Int,
    b :: Array{T, 3}
)
```

Documentation for the other ``Method`` object.

---

With some heuristic to determine when we split a signature over multiple lines.